### ISSUE TITLE
[Bug]: entered date is shifted back one day in Birthday

### ISSUE THREAD

**Issue by alexanderdd** [nextcloud/server#49828]
### Steps to reproduce

1. (not sure if that is relevant) be located in Brazil (with correct timezone setting of computer/browser), NC server is in Germany 
2. in settings > personal info, set birthday date to (example) 31.1.1986
3. birthday is displayed as 30.1.1986

### Expected behavior

entered date = displayed date

### Nextcloud Server version

30

### Additional info

this is on c.nc.com - can't check most of the other needed info

maybe related? https://github.com/nextcloud/calendar/issues/6528

same for settings > Availability > Absence  #61584

**Comment by ChristophWurst** [comment]
This sounds similar to https://github.com/nextcloud/server/issues/42139.

**Comment by SebastianKrupinski** [comment]
Tested. Confirmed.

### Issue

The issue is the way that javascript new Date() functions

https://github.com/nextcloud/server/blob/c21888e6ae1a267588fb110ec27cfc6e3b1f484e/apps/dav/src/components/AbsenceForm.vue#L84

The date is sent from the backend to the frontend as a string "2024-12-31", without time or timezone. Javascript Date() creates the date in UTC timezone because there is no timezone in the string, then is applies the browsers time zone to the UTC date. This then causes the date to roll back one day, if the browser timezone has a positive offset (grater than UTC).

Example:

2024-12-31 -> in UTC time zone is 2024-12-31 T00:00:00

2024-12-31 -> in America/Toronto time zone is 2024-12-30 T19:00:00

2024-12-31 -> in Australia/Sydney time zone is 2024-12-31 T11:00:00

### Solution

- Send apply timezone and full date/time string with time zone from backend
- Send time zone from backend and calculate date with timezone offset

**Comment by alexanderdd** [comment]
Still a problem in NC 31.

**Comment by gavyncaldwell** [comment]
Yep likewise experiencing this issue.

**Comment by solonovamax** [comment]
yeah it seems like the dates are not being properly localized into the user's timezone.

For example for me, updating my birthday makes a `PUT` request with `{"key":"birthdate","value":"2003-02-07T00:00:00.000Z"}`. However, I have configured my timezone to be `America/Toronto`, so the birthday *should* be localized to the EST timezone, and it should instead be making a `PUT` request with `{"key":"birthdate","value":"2003-02-07T00:00:00.000-05:00"}`.

If I manually make a `PUT` request with `curl` with the correct date, then after reloading it shows the correct birthday in the UI and no longer shows "Happy birthday solo" on the dashboard on 02-06 (when my birthday is set to 02-07).

**Comment by SebastianKrupinski** [comment]
Hi @solonovamax 

Are you talking about the birth date in the user profile settings? Or in a Contact?

**Comment by odzhychko** [comment]
I create a separate issue for the Availability settings in https://github.com/nextcloud/server/issues/61584.

@SebastianKrupinski This is about the birthday in "personal info" settings and not about birthday inputs in contacts. (`{"key":"birthdate", ...}` is the API used by the settings.)
